CONTRIBUTING.md

```diff
@@ -163,3 +163,154 @@ minor versions at this time.
 Examples:
 
 - Refactored how Checks are stored in Etcd, `sensu-backend migrate` is required to upgrade
+
+## Development
+
+Sensu is written in Go, and targets the latest stable release of the Go
+compiler. To work on Sensu, you will need the latest release of Go.
+
+[Go installation instructions](https://golang.org/doc/install)
+
+### Protobuf
+
+#### Overview
+
+We are using the version **proto3** of the protocol buffers language. Here are some useful resources:
+
+[To learn more about protocol buffers](https://developers.google.com/protocol-buffers/docs/overview)
+
+[The proto3 language guide](https://developers.google.com/protocol-buffers/docs/proto3)
+
+#### Installation
+
+Install the protobuf compiler since we don't use the one that golang uses.
+```
+brew install protobuf
+```
+Otherwise, see the **for non-C++ users** [instructions here.](https://github.com/google/protobuf#protocol-compiler-installation)
+
+#### Quick Start
+
+Once you make a change to any `*.proto` file within the **types** package, you will need to regenerate the associated `*.pb.go` file. To do so, simply run `go generate` on the package.
+
+### Dependencies
+
+Sensu uses [golang/dep](https://github.com/golang/dep) for managing its
+dependencies. You will need to install the latest stable version of dep in
+order to install Sensu's dependencies.
```

Comment:
You don't really need `dep` if you don't plan on changing what we vendor since... we vendor it!